torchvision/datapoints/_dataset_wrapper.py

```diff
@@ -98,7 +97,16 @@ def wrap_dataset_for_transforms_v2(dataset, target_keys=None):
             f"but got {target_keys}"
         )
 
-    return VisionDatasetDatapointWrapper(dataset, target_keys)
+    return type(
+        f"Wrapped{type(dataset).__name__}",
+        (type(dataset),),
+        {
+            "__init__": wrapped_init,
+            "__getattr__": wrapped_getattr,
+            "__getitem__": wrapped_getitem,
+            "__len__": wrapped_len,
```

Comment:
Why do we need this? Shouldn't the MRO do its job without us having to override anything here?

Reply:
Oh, if it's because we have now removed the `VisionDatasetDatapointWrapper`, let's just not remove it please. At least not in this PR. (we can keep it with a leading underscore if we want to).